src/app/clusters/thermostat-server/thermostat-server.cpp

```diff
@@ -297,7 +297,7 @@ MatterThermostatClusterServerPreAttributeChangedCallback(const app::ConcreteAttr
             return imcode::InvalidValue;
         if (AutoSupported)
         {
-            if (requested < OccupiedHeatingSetpoint + DeadBandTemp)
+            if (requested > OccupiedHeatingSetpoint + DeadBandTemp)
```

Comment:
This change is wrong, as far as I can tell.  What the spec says is:

> if, and only if, the AUTO feature is supported, a deadband must be maintained between Heating and Cooling setpoints and limits:
> ...
> OccupiedHeatingSetpoint ≤ (OccupiedCoolingSetpoint - MinSetpointDeadBand)

So if that condition is violated, i.e. "OccupiedHeatingSetpoint + MinSetpointDeadBand > OccupiedCoolingSetpoint", then that's not a valid setting.  And that's what this code is checking for.

Please clearly explain why you are introducing these changes that violate the spec (and incidentally fail the conformance tests like Test_TC_TSTAT_2_2).